frontend/src/services/distance.js

```diff
@@ -0,0 +1,42 @@
+const distanceKm = (lat1, lon1, lat2, lon2) => {
+  if ((lat1 === lat2) && (lon1 === lon2)) {
+    return 0;
+  }
+
+  const radlat1 = (Math.PI * lat1) / 180;
+  const radlat2 = (Math.PI * lat2) / 180;
+  const theta = lon1 - lon2;
+  const radtheta = (Math.PI * theta) / 180;
+  let dist = Math.sin(radlat1) * Math.sin(radlat2) + Math.cos(radlat1) * Math.cos(radlat2) * Math.cos(radtheta);
+  if (dist > 1) {
+    dist = 1;
+  }
+  dist = Math.acos(dist);
+  dist = (dist * 180) / Math.PI;
+  dist = dist * 60 * 1.1515;
+  return dist * 1.609344;
```

Comment:
as this is proportional, if you want to speed it up you can remove it

Reply:
Hmm, I'd rather not. If we want to print the distance in the future or to better deal with precision, I prefer to keep it in km.